ipydatawidgets/ndarray/union.py

```diff
@@ -36,6 +37,24 @@ def __init__(self, default_value=Undefined, dtype=None, shape_constraint=None,
         self._registered_validators = {}
         self._registered_observer = {}
 
+    def set(self, obj, value):
+        new_value = self._validate(obj, value)
+        try:
+            old_value = obj._trait_values[self.name]
+        except KeyError:
+            old_value = self.default_value
+
+        obj._trait_values[self.name] = new_value
+        try:
+            silent = np.array_equal(old_value, new_value)
+        except:
+            # if there is an error in comparing, default to notify
+            silent = False
+        if silent is not True:
```

Comment:
This should probably be `!= True`, as this also gives deprecation warnings for Python 3.8 and above

Reply:
Hi @vidartf ! Thank you for your quick answer.
I'm not sure to understand your comment, is there something going wrong?
Could you please give me more details?

It works fine for me on Python3.8, running this simple example:

``` python
import warnings

from ipydatawidgets import DataUnion
from traitlets import HasTraits, observe
import numpy as np

warnings.simplefilter("always")

class TestClass(HasTraits):

    array = DataUnion()

    @observe('array')
    def print(self, changes):
        print(changes['new'])

test = TestClass()
test.array = np.array([10., 20., 30.], dtype=np.float32)
test.array = np.array([10., 40., 30.], dtype=np.float32)  # <-- was ok, still ok
test.array = np.array([10., 40., 30.], dtype=np.float32)  # <-- observer no more triggered when array is unchanged
test.array = np.array([10., 40.], dtype=np.float32)  # <-- no more DeprecationWarning
```

Reply:
Sorry, my mistake. Python 3.8 has a new warning for comparing literals with `is`, and I got confused and thought it included boolean literals as well. It does not :)